src/test/ui/nll/maybe-initialized-drop.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//compile-flags: -Z emit-end-regions -Z borrowck-mir -Z nll
+
+struct Wrap<'p> { p: &'p mut i32 }
+
+impl<'p> Drop for Wrap<'p> {
+    fn drop(&mut self) {
+        *self.p += 1;
+    }
+}
+
+fn foo() {
+    let mut x = 0;
+    let wrap = Wrap { p: &mut x };
+    x += 1; //~ ERROR because of dtor
+}
+
+fn bar() {
+    let mut x = 0;
+    let wrap = Wrap { p: &mut x };
+    std::mem::drop(wrap);
+    x += 1; // OK, drop is inert
+}
+
+struct Foo<'p> { a: String, b: Wrap<'p> }
+
+fn move_string(_a: String) { }
+
+fn move_wrap<'p>(_b: Wrap<'p>) { }
+
+fn baz_a() {
+    let mut x = 0;
+    let wrap = Wrap { p: &mut x };
+    let s = String::from("str");
+    let foo = Foo { a: s, b: wrap };
+    move_string(foo.a);
+    x += 1; //~ ERROR because of Foo.b's Wrap dtor
+}
+
+fn baz_a_b() {
+    let mut x = 0;
+    let wrap = Wrap { p: &mut x };
+    let s = String::from("str");
+    let foo = Foo { a: s, b: wrap };
+    move_string(foo.a);
+    move_wrap(foo.b);
+    x += 1; // OK, drops are inert
```

Comment:
@nikomatsakis is this comment right? Does it make sense?

Reply:
That comment seems correct to me.